src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs

```diff
@@ -308,7 +308,9 @@ internal static bool IsTypeDefinition(RuntimeType type)
                 Assembly a;
                 try
                 {
+#pragma warning disable SYSLIB0018 // ReflectionOnly loading is not supported and throws PlatformNotSupportedException.
                     a = Assembly.ReflectionOnlyLoad(an);
```

Comment:
Not sure if mono is used in lower versions, does not look like it is, if that is the case we probably should just throw PNSE in this `reflectionOnly` case cc @marek-safar

Reply:
Why do you think it's Mono specific? I think we should bubble up this information not hide it if always throws

Reply:
> Why do you think it's Mono specific? 

It didn't warn in CoreClr corelib, I don't even see this method having ` bool reflectionOnly` parameter in CoreClr

>  I think we should bubble up this information not hide it if always throws

It is not throwing unconditionally, as you know we cannot conditionally Obsolete an API

Anyway, I see this API always called with a value of `false` for `bool reflectionOnly` parameter, we might want to remove this parameter from this method, probably with a different PR

Reply:
> I see this API always called with a value of `false` for `bool reflectionOnly` parameter, we might want to remove this parameter and related code section from this method, probably with a different PR

Sounds good to me.

Reply:
Created an issue for this https://github.com/dotnet/runtime/issues/51838